staging/src/k8s.io/apiextensions-apiserver/test/integration/finalization_test.go

```diff
@@ -163,3 +164,29 @@ func TestFinalizationAndDeletion(t *testing.T) {
 		t.Fatalf("unable to delete crd: %v", err)
 	}
 }
+
+func TestApplyCRDuringCRDFinalization(t *testing.T) {
+	tearDown, apiExtensionClient, dynamicClient, err := fixtures.StartDefaultServerWithClients(t)
+	require.NoError(t, err)
+	defer tearDown()
+
+	// Create a CRD with a finalizer which will stall deletion
+	noxuDefinition := fixtures.NewNoxuV1CustomResourceDefinition(apiextensionsv1.ClusterScoped)
+	noxuDefinition.SetFinalizers([]string{"noxu.example.com/finalizer"})
+	noxuDefinition, err = fixtures.CreateNewV1CustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+	require.NoError(t, err)
+
+	// Delete the CRD. Since it has a finalizer it will be stuck in terminating state
+	err = apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().Delete(t.Context(), noxuDefinition.Name, metav1.DeleteOptions{})
+	require.NoError(t, err)
+
```

Comment:
the create-via-SSA attempt below can race the CRD delete becoming effective (handler setup observing the delete and adjusting the handlers to inject the fail-on-create admission)

make the apply a dry-run so that it won't actually persist if it gets through admission, and put it in a loop that retries until it gets the expected error or we get a test timeout:

```go
err := wait.PollImmediate(100*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
  _, err := noxuResourceClient.Apply(t.Context(), name, instance, metav1.ApplyOptions{DryRun: []string{"All"}, FieldManager: "manager"})
  if err == nil {
    t.Log("apply was not blocked, retrying...")
    return false, nil
  }
  return true, err
})
wantErr := ...
require.Error(t, fmt.Errorf("%s", wantErr))
```